### PR TITLE
#160451559 Implement Social Authentication 

### PR DESCRIPTION
**Pull request by andela-phemmy**
#### What does this PR do?
     This PR implements Social authentication functionality
#### Description of Task to be completed?
      1. Add environment variables for social auth URLs
      2. Add social auth href to all social auth buttons
      3. Test SocialSignIn, LoginTypeSelectors and LoginType components
#### How should this be manually tested?
      On localhost
        1. start the app on port `8080`
        2. Navigate to sign in
        3. Click on any social auth button
        4. Or navigate to sign up
        5. Click on any social auth button
        6. In any of the above cases, you will be redirected to the social auth page
        7. Authenticate with your credentials and proceed to landing page 
#### Any background context you want to provide?
        N/A
### What are the relevant pivotal tracker stories?
       #160451559
#### Screenshots (if appropriate)
screen shot 2018-11-14 at 12 15 42 pm

#### Questions:
        N/A

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ {
+     "singleQuote": true,
+     "trailingComma": "es5",
+     "jsxSingleQuote": true,
+     "jsxBracketSameLine": true
+ }
+
```
Comment (e0works) [inline .prettierrc]: You forgot to add a new line, here

File: src/action/resetPasswordAction.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// third-party importation
+import http from 'axios';
+
+// constants importation
```
Comment (e0works) [inline src/action/resetPasswordAction.js]: I think this comment should read
```js
// action types
```

File: src/components/SocialLogin/SocialSignIn.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { Fragment } from 'react';
```
Comment (e0works) [inline src/components/SocialLogin/SocialSignIn.jsx:2]: Please, document the Component `SocialSignIn`
also put a comment on the react library importation

File: src/components/__tests__/forgot.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// react and react-redux libraries
```
Comment (e0works) [inline src/components/__tests__/forgot.test.js]: I think you should separate this 2 and 3.
It should read

```js
// react libraries
import React from 'react';

// third-party libraries
import { Provider } from 'react-redux';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
```

File: src/components/__tests__/reset.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// react libraries
```
Comment (e0works) [inline src/components/__tests__/reset.test.js]: Please comment on the imports properly
react-redux is not part of react library

**Review by e0works — CHANGES_REQUESTED** [review]

**Review by e0works — APPROVED** [review]

File: src/components/SocialLogin/SocialSignIn.jsx [hunk 6]
```diff
@@ -1,17 +1,21 @@
...
+/**
+ * @desc renders social login group
+ */
 const SocialSignIn = () => (
   <Fragment>
     <div className='social-wrap text-center'>
       <p>Social sign-in</p>
       <div className='social-inner d-flex justify-content-center'>
-        <a href='#'>
+        <a href={process.env.FACEBOOK_URL}>
```
Comment (KaiserPhemi) [inline src/components/SocialLogin/SocialSignIn.jsx:12]: #### Critical
You could extract all the urls once and just pass them at the anchor tags i.e:
```js
const { // needed props } = process.env;
```

File: src/components/__tests__/loginType.test.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import React from 'react';
+
+// enzyme third party libraries
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+// component
+import LoginType from '../login/LoginType';
+
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (KaiserPhemi) [inline src/components/__tests__/loginType.test.js:11]: #### Critical
Refactor this i.e. the `enzyme` setup should go into a file e.g. `testSetup` file then add the key to your `jest` configuration in `package.json`.

Also, do this for all the test files.

For more clarity check [Using `enzyme` with React16](https://airbnb.io/enzyme/docs/installation/react-16.html) and [`enzyme` setup](https://stackoverflow.com/questions/46627353/where-should-the-enzyme-setup-file-be-written)

File: src/components/login/LoginTypeSelector.jsx [hunk 8]
```diff
@@ -18,14 +18,17 @@ const LoginTypeSelector = props => (
         <LoginType
             className={'fab fa-facebook'}
             loginTypeName={'Facebook'}
+            href={process.env.FACEBOOK_URL}
```
Comment (KaiserPhemi) [inline src/components/login/LoginTypeSelector.jsx:21]: #### Critical
Same as above. Extract the urls from the global env object first.

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
@andela-phemmy 
Implement feedback.

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**